### PR TITLE
fix luis-cross train does not find imported files with locale

### PR DESCRIPTION
**Pull request by cosmicshuai**
closes: #1205 

In this PR, the import resolver for luis and qna files will find the correct import files based on the locale info.
Suppose we have a.en-us.lu with content
```
[import](b.lu)
```
The import resolver will find the b.en-us.lu

### CODE REVIEW

File: packages/lu/src/parser/cross-train/cross-train.js [hunk 1]
```diff
@@ -19,14 +19,23 @@ module.exports = {
...
-    const qnaContents = await filehelper.getFilesContent(input, fileExtEnum.QnAFile)
+    const luContents = await filehelper.getFilesContent(input, fileExtEnum.LUFile, ignoredFolders)
+    const qnaContents = await filehelper.getFilesContent(input, fileExtEnum.QnAFile, ignoredFolders)
     const configContent = await filehelper.getConfigContent(config)
+    const defaultLocale = 'en-us'
 
     let importResolver = async function (id, idsToFind) {
       let importedContents = []
+      const idWithoutExt = path.basename(id, path.extname(id))
+      const locale = /\w\.\w/.test(idWithoutExt) ? idWithoutExt.split('.').pop() : defaultLocale;
```
Comment (feich-ms) [inline packages/lu/src/parser/cross-train/cross-train.js:38]: >const locale = /\w\.\w/.test(idWithoutExt) ? idWithoutExt.split('.').pop() : defaultLocale; [](http://example.com/codeflow?start=6&length=91)

This is not always true, for instance, a.xxx.lu, your logic here will make locale be xxx.

File: packages/luis/test/fixtures/testcases/testImportWithLocale/bin/assistant_core.en-us.lu [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# BotTour
+- bot tour
+- onboarding
+- get started
+- start the tour
+- what can you do
+- what can i ask you
```
Comment (feich-ms) [inline packages/luis/test/fixtures/testcases/testImportWithLocale/bin/assistant_core.en-us.lu:7]: I think just one file is enough

**Review by feich-ms — APPROVED** [review]
:shipit:

File: packages/luis/test/fixtures/testcases/testImportWithLocale/bin/assistant_core.en-us.lu (hunk 2 above)
Comment (cosmicshuai) [reply packages/luis/test/fixtures/testcases/testImportWithLocale/bin/assistant_core.en-us.lu]: modified

**Review by boydc2014 — APPROVED** [review]

File: packages/cli/README.md [hunk 3]
```diff
@@ -891,6 +891,8 @@ OPTIONS
   --[no-]intra-dialog      Only do intra dialog cross train
 
   --log                    Writes out log messages to console
+
+  --ignore                ignore given folders under the input directory, for example, --ignore bin,obj,lib, this will ignore the /bin, /obj, /lib folders under the input path
```
Comment (gabog) [inline packages/cli/README.md:895]: can we name the paramater exclude instead of ignore

**Comment by boydc2014** [comment]
@munozemilio this is a fix for adaptive runtime CI/CD, it's been verified, can you help get it in

File: packages/cli/README.md (hunk 3 above)
Comment (cosmicshuai) [reply packages/cli/README.md:895]: right, that's more accurate. I changed it to `exclude`

Comment (gabog) [inline packages/cli/README.md:895]: Hi, I think you missed the readmes (I see ignore is still being used here and in other places)

**Review by gabog — DISMISSED** [review]
I think you missed changing the ignore parameter in the readme files (I still see --ignore there and should be --exclude, right?)

File: packages/cli/README.md (hunk 3 above)
Comment (cosmicshuai) [reply packages/cli/README.md]: right. I've changed the parameter name in two readme files

File: packages/luis/src/commands/luis/cross-train.ts [hunk 4]
```diff
@@ -22,7 +22,8 @@ export default class LuisCrossTrain extends Command {
     force: flags.boolean({char: 'f', description: 'If --out flag is provided with the path to an existing file, overwrites that file', default: false}),
     log: flags.boolean({description: 'Writes out log messages to console', default: false}),
     'inner-dialog': flags.boolean({description: 'Only do inner dialog cross train', default: true, allowNo: true}),
-    'intra-dialog': flags.boolean({description: 'Only do intra dialog cross train', default: true, allowNo: true})
+    'intra-dialog': flags.boolean({description: 'Only do intra dialog cross train', default: true, allowNo: true}),
+    exclude: flags.string({description: 'Excludes folders under the input directory, seperated by ",". If not specified, all luis and qna files will be included in the cross-train'})
```
Comment (munozemilio) [inline packages/luis/src/commands/luis/cross-train.ts:26]: typo: 'separated'

File: packages/luis/README.md [hunk 5]
```diff
@@ -431,6 +431,8 @@ OPTIONS
   --[no-]intra-dialog      Only do intra dialog cross train
 
   --log                    Writes out log messages to console
+
+  --excluder                Excludes given folders under the input directory, for example, --exclude bin,obj,lib, this will ignore the /bin, /obj, /lib folders under the input path
```
Comment (munozemilio) [inline packages/luis/README.md:435]: exclude ?

**Review by munozemilio — CHANGES_REQUESTED** [review]
Should content be added to the test empty files? Some typos also need to be addressed.

File: packages/luis/src/commands/luis/cross-train.ts (hunk 4 above)
Comment (cosmicshuai) [reply packages/luis/src/commands/luis/cross-train.ts]: fixed

File: packages/luis/README.md (hunk 5 above)
Comment (cosmicshuai) [reply packages/luis/README.md]: fixed

**Comment by cosmicshuai** [comment]
@munozemilio The empty content test files are from a real Composer bot, I think it is OK to keep them empty. There are import files with locale. So the test covers the code added in this PR.

File: packages/luis/README.md [hunk 6]
```diff
@@ -431,6 +431,8 @@ OPTIONS
   --[no-]intra-dialog      Only do intra dialog cross train
 
   --log                    Writes out log messages to console
+
+  --exclude                Excludes given folders under the input directory, for example, --exclude bin,obj,lib, this will ignore the /bin, /obj, /lib folders under the input path
```
Comment (munozemilio) [inline packages/luis/README.md:435]: FYI This will be overriden by the doc script.

**Review by munozemilio — APPROVED** [review]

**Pull request merged**